pandas/tests/indexing/multiindex/test_multiindex.py

```diff
@@ -206,3 +206,20 @@ def test_multiindex_with_na_missing_key(self):
         )
         with pytest.raises(KeyError, match="missing_key"):
             df[[("missing_key",)]]
+
+    def test_multiindex_dtype_preservation(self):
+        # GH51261
+        columns = MultiIndex.from_tuples([("A", "B")], names=["lvl1", "lvl2"])
+        df = DataFrame(["value"], columns=columns).astype("category")
+        assert (df["A"].dtypes == "category").all()
+
+        # geopandas 1763 analogue
+        df = DataFrame(
+            [[1, 0], [0, 1]],
+            columns=[
+                ["foo", "foo"],
+                ["location", "location"],
+                ["x", "y"],
+            ],
+        ).assign(bools=Series([True, False], dtype="boolean"))
+        assert df["bools"].dtype == "boolean"
```

Comment:
Could you use `pandas.core.dtypes.common.is_bool_dtype(df["bools"])` here?

Reply:
For this case we actually want to be more explicit to check the extension dtype (`is_bool_dtype` would also pass with numpy bool dtype). Possible alternative is `isinstance(.., BooleanDtype)`